buffer/pool.go

```diff
@@ -156,26 +155,22 @@ func (b *Buffer) Size() int {
 
 // DumpTo outputs the contents of a buffer to a writer and resets the buffer.
 func (b *Buffer) DumpTo(w io.Writer) (written int, err error) {
-	var n int
-	for _, buf := range b.bufs {
-		if err == nil {
-			n, err = w.Write(buf)
-			written += n
-		}
-		putBuf(buf)
+	bufs := (net.Buffers)(b.bufs)
```

Comment:
Maybe change b.bufs type to net.Buffers?

Reply:
I did not want to have this change impact unrelated parts of the code, and I could not see a benefit in changing it (compared to simply casting it).

If you really prefer it I can do it, I don't think it should be hard to change.